packages/core/src/app/shipping/PayPalFastlaneShippingAddress.tsx

```diff
@@ -20,22 +47,99 @@ export const PayPalFastlaneShippingAddress: FC<PayPalFastlaneShippingAddressProp
         onUnhandledError,
         initialize,
         deinitialize,
-        isLoading,
         shippingAddress,
+        addresses,
+        handleFieldChange,
     } = props;
+    const [isLoading, setIsLoading] = useState<boolean>(true);
 
-    return (
-        <PayPalFastlaneShippingAddressForm
-            address={shippingAddress}
-            countries={countries}
-            deinitialize={deinitialize}
-            formFields={formFields}
-            initialize={initialize}
-            isLoading={isLoading}
-            methodId={methodId}
-            onAddressSelect={onAddressSelect}
-            onFieldChange={onFieldChange}
-            onUnhandledError={onUnhandledError}
-        />
+    const paypalFastlaneShippingComponent = useRef<PayPalFastlaneAddressComponentRef>({});
+    const fastlaneOptions = (provider: string) => {
+        return {
+            [provider]: {
+                onPayPalFastlaneAddressChange: (
+                    showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+                ) => {
+                    paypalFastlaneShippingComponent.current.showAddressSelector =
+                        showPayPalFastlaneAddressSelector;
+                },
+            },
+        };
+    }
+
+    const initializationOptions: ShippingInitializeOptions = isPayPalCommerceFastlaneMethod(
+        methodId,
     )
+        ? fastlaneOptions('paypalcommercefastlane')
+        : fastlaneOptions('braintreefastlane');
+
+    const initializeShippingStrategyOrThrow = async () => {
+        try {
+            await initialize({
+                methodId,
+                ...initializationOptions,
+            });
+        } catch (error) {
+            if (typeof onUnhandledError === 'function' && error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    const deinitializeShippingStrategyOrThrow = async () => {
+        try {
+            await deinitialize({ methodId });
+        } catch (error) {
+            if (typeof onUnhandledError === 'function' && error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+        setIsLoading(false);
```

Comment:
I guess this method call should be moved to `initializeShippingStrategyOrThrow` to avoid infinite loading in this component